packages/vite/src/node/config.ts

```diff
@@ -66,7 +67,13 @@ export function defineConfig(config: UserConfigExport): UserConfigExport {
   return config
 }
 
-export type PluginOption = Plugin | false | null | undefined | PluginOption[]
+export type PluginOption =
+  | Plugin
+  | Promise<Plugin>
```

Comment:
The promise options also need typeof `false | null | undefined | PluginOption[]`?

Reply:
Good point, we should probably wrap all those as a promisable too.